### ISSUE TITLE
flatten ignores abort==true

### ISSUE THREAD

**Issue by regular** [dominictarr/pull-stream#29]
This is causing superfluous reads.

## Example code

``` js
test('flatten - number of reads', function (t) {
  var reads = 0
  pull(
    pull.values([
      pull.values([1, 2, 3]),
    ]),
    pull.flatten(),
    pull.through(function() {
      reads++
      console.log('READ', reads)
    }),
    pull.take(2),
    pull.collect(function (err, numbers) {
      t.deepEqual([1, 2], numbers)
      t.equal(reads, 2)
      t.end()
    })
  )
}) // ==> fails 3 != 2
```

**Comment by dominictarr** [comment]
cool, just trying to figure out if this should be a minor or a breaking change?

**Comment by dominictarr** [comment]
I'm gonna consider this a feature (minor) change, since no the way take read one more was a bug not a feature. This change is backwards compatible with that, except for when you call it with a number, in which case it actually does what you'd expect it to do.

**Comment by dominictarr** [comment]
merged into 2.28.0
